app_dart/test/request_handlers/github_webhook_test.dart

```diff
@@ -2012,6 +2021,69 @@ void foo() {
     });
   });
 
+  group('github webhook create branch event', () {
+    setUp(() {
+      request.headers.set('X-GitHub-Event', 'create');
+      request.body = generateCreateBranchEvent('flutter-2.12-candidate.4', 'master', 'flutter/flutter');
+      final Uint8List body = utf8.encode(request.body!) as Uint8List;
+      final Uint8List key = utf8.encode(keyString) as Uint8List;
+      final String hmac = getHmac(body, key);
+      request.headers.set('X-Hub-Signature', 'sha1=$hmac');
+    });
+    test('should add branch to db if db is empty', () async {
+      expect(db.values.values.whereType<Branch>().length, 0);
+      await tester.post(webhook);
+
+      expect(db.values.values.whereType<Branch>().length, 1);
+      final Branch branch = db.values.values.whereType<Branch>().single;
+      expect(branch.repository, 'flutter/flutter');
+      expect(branch.branch, 'flutter-2.12-candidate.4');
+      expect(branch.defaultBranch, 'master');
+    });
+
+    test('should not add duplicate entity if branch already exists in db', () async {
```

Comment:
You can add mock data in the fake db as it essentially is a map. This way you do not need to have two fake requests.